apps/arweave/src/ar_tx_poller.erl

```diff
@@ -0,0 +1,160 @@
+-module(ar_tx_poller).
+-behaviour(gen_server).
+
+-export([
+	start_link/0
+]).
+
+-export([
+	init/1,
+	handle_call/3,
+	handle_cast/2,
+	handle_info/2,
+	terminate/2
+]).
+
+-include("../include/ar.hrl").
+-include("../include/ar_config.hrl").
+-record(state, {
+	last_seen_tx_timestamp = 0,
+	pending_txids = []
+}).
+
+%% Number of peers to query for a transaction.
+-define(QUERY_PEERS_COUNT, 5).
+
+%% Check interval in milliseconds - how long to wait before polling
+%% since the last transaction push. If the node is not public (so it
+%% never receives transactions by push), we wait this long starting from
+%% the moment we join the network only once and then keep polling
+%% for transactions more frequently.
+-ifdef(AR_TEST).
+-define(CHECK_INTERVAL_MS, 5_000).
+-else.
+-define(CHECK_INTERVAL_MS, 30_000).
+-endif.
+
+%% Poll interval in milliseconds - how long we wait before downloading a new
+%% transaction or polling the mempools for new transactions.
+-ifdef(AR_TEST).
+-define(POLL_INTERVAL_MS, 500).
+-else.
+-define(POLL_INTERVAL_MS, 200).
+-endif.
+
+%%% Public API.
+
+start_link() ->
+	gen_server:start_link({local, ?MODULE}, ?MODULE, [], []).
+
+%%% Gen server callbacks.
+
+init([]) ->
+    [ok, ok] = ar_events:subscribe([tx, node_state]),
+	{ok, #state{}}.
+
+handle_call(Request, From, State) ->
+	?LOG_WARNING("Unexpected call: ~p from ~p", [Request, From]),
+	{reply, ignored, State}.
+
+handle_cast(check_for_received_txs, State) ->
+	%% Check if there have been any transactions received in the last
+	%% ?CHECK_INTERVAL_MS milliseconds.
+	TimestampDiff = erlang:system_time(microsecond) - State#state.last_seen_tx_timestamp,
+	State3 =
+		case TimestampDiff > 0 andalso TimestampDiff > ?CHECK_INTERVAL_MS of
+			true ->
+				State2 = check_for_received_txs(State),
+				ar_util:cast_after(?POLL_INTERVAL_MS, self(), check_for_received_txs),
```

Comment:
Actually one more question: It looks like the node will only query at most one tx every ?POLL_INTERVAL_MS, is that correct/by design? I.e. once the CHECK_INVTERVAL_MS has elapsed, the node will query the mempool of the a peer, and the request one tx at a time at most every ?POLL_INTERVAL_MS seconds (rather then just request them sequentially as fast as it can.)

Reply:
Yes, I should probably make it requests txs without delays

Reply:
https://github.com/ArweaveTeam/arweave/pull/714/commits/e99067efb355b0ec939791ea7526be57d85f5370